src/number_theory/divisors.lean

```diff
@@ -398,4 +403,86 @@ begin
   exact nat.lt_succ_of_le (nat.divisor_le (nat.mem_divisors.2 ⟨ha, h⟩))
 end
 
+
+
+/-- # Prime divisors -/
+
+lemma prime_divisors_mem {n p : ℕ} (hn : n ≠ 0) : p ∈ n.prime_divisors ↔ p ∣ n ∧ prime p :=
+by { unfold prime_divisors, rw [mem_filter, mem_divisors, and_iff_left hn] }
+
+lemma prime_of_mem_prime_divisors {n p : ℕ} (h : p ∈ n.prime_divisors) : prime p :=
+by { rw [prime_divisors, mem_filter] at h, exact h.2 }
+
+lemma dvd_of_mem_prime_divisors {n p : ℕ} (h : p ∈ n.prime_divisors) : p ∣ n :=
+by { rw [prime_divisors, mem_filter] at h, exact dvd_of_mem_divisors h.1 }
+
+/-- 0 has no prime divisors -/
+lemma prime_divisors_zero : prime_divisors 0  = (∅ : finset ℕ) :=
+filter_empty _
+
+/-- 1 has no prime divisors -/
+lemma prime_divisors_one : prime_divisors 1 = (∅ : finset ℕ) :=
+filter_singleton prime 1
+
+lemma prime_divisors_eq_to_finset_factors (n : ℕ) : n.prime_divisors = n.factors.to_finset :=
```

Comment:
if `factors` exists, why do we need this? is it for those "setty" lemmas? why not just make an api for `n.factors.to_finset`?

Reply:
Yeah, this crossed my mind too. Certainly we need a lemma that says `n.factors.to_finset = (finset.range (n+1)).filter prime`, but I don't know if we actually need a special name for this operation

Reply:
Mainly I proved `prime_divisors_eq_to_finset_factors` just because Ruben Van de Velde asked if they were equal [in the Zulip thread](https://leanprover.zulipchat.com/#narrow/stream/113489-new-members/topic/Proof.20of.20Euler's.20product.20formula.20for.20totient/near/261380440).  I'm not using this lemma for anything at the moment. 

I wondered in that thread whether it would be better to change the definition of `prime_divisors` to `factors.to_finset`, but I didn't want to make the change just speculatively and then have to tidy up.  But if this would be a better definition — if it's more efficient or in some other way substantially nicer than `(divisors n).filter prime` — then I'm happy to change it.

Reply:
So would you suggest just taking out `prime_divisors_eq_to_finset_factors`?

Reply:
Ok, I've taken out `prime_divisors_eq_to_finset_factors`, which I wasn't using, and replaced it with the following, which I've found useful in some other work elsewhere:
```
lemma factor_of_prime_divisor {p n : ℕ} (hp : p ∈ n.prime_divisors) : p ∈ n.factors
```

Reply:
Sorry, I don't think either of us were asking for you to remove the lemma; rather, the lemma was the hint to us that `prime_divisors` _maybe_ shouldn't exist at all.

Reply:
Ah, I see, because the lemma reveals that it's essentially just an alias for `factors.to_finset`, and so the properties satisfied by `prime_divisors` should follow straightforwardly from the corresponding properties of `factors`.

Reply:
precisely, just rewrite the lemmas that don't already exist for `n.factors`/`n.factors.to_finset` instead of `prime_divisors`.  I think everything up to `coprime_prime_divisors_disjoint` should even generalize to the list setting.

Reply:
Great, thanks